app/controllers/tasks.js

```diff
@@ -1,67 +1,61 @@
 import Controller from '@ember/controller';
 import { tracked } from '@glimmer/tracking';
-import { action, set } from '@ember/object';
+import { action } from '@ember/object';
 import ENV from 'website-my/config/environment';
-import TASK_STATUS from 'website-my/constants/tasks';
+import { TASK_STATUSES } from 'website-my/constants/tasks';
 
 const API_BASE_URL = ENV.BASE_API_URL;
-const { ACTIVE, BLOCKED, COMPLETED, PENDING } = TASK_STATUS;
 
 export default class TasksController extends Controller {
-  @tracked showDropDown = false;
+  taskStatuses = TASK_STATUSES;
+  DEFAULT_TASK_TYPE = this.taskStatuses[0];
+  @tracked showDropDown = true;
   @tracked taskFields = {};
   @tracked allTasks = this.model;
+  @tracked isLoading = false;
+  @tracked userSelectedTask = this.DEFAULT_TASK_TYPE;
 
   @action toggleDropDown() {
     this.showDropDown = !this.showDropDown;
   }
 
   @tracked tasksByStatus = {};
-  taskStatusList = [ACTIVE, BLOCKED, PENDING];
 
-  filterTasksByStatus(allTasks, status) {
-    return allTasks.filter((task) => task.status === status);
-  }
-
-  seperateTasksByStatus() {
-    this.taskStatusList.forEach((status) => {
-      this.tasksByStatus[status] = this.filterTasksByStatus(
-        this.allTasks,
-        status
+  calculateTasksToShowBySelectedStatus() {
+    if (this.userSelectedTask.key === 'all') {
+      this.tasksToShow = this.allTasks;
+    } else {
+      this.tasksToShow = this.allTasks.filter(
+        (task) => task.status === this.userSelectedTask.key
       );
-    });
+    }
   }
 
   cleanReqBody(object) {
-    const cleanReqObj = {
-      ...object,
-      percentCompleted: parseInt(object.percentCompleted),
-    };
-    return cleanReqObj;
+    const taskCompletionPercentage = object.percentCompleted;
+    if (taskCompletionPercentage) {
+      object.percentCompleted = parseInt(taskCompletionPercentage);
+    }
+    return object;
   }
 
-  defaultTaskType = ACTIVE;
-  @tracked tasksToShow = this.allTasks.filter(
-    (task) => task.status === this.defaultTaskType
-  );
-
-  @action toggleTasks(taskType) {
-    this.seperateTasksByStatus();
-    this.tasksToShow = this.tasksByStatus[taskType];
+  @action changeUserSelectedTask(statusObject) {
+    this.userSelectedTask = statusObject;
+    this.calculateTasksToShowBySelectedStatus();
   }
 
+  @tracked tasksToShow = this.allTasks;
+
   @action onTaskChange(key, value) {
     this.taskFields[key] = value;
   }
 
   @action async handleUpdateTask(taskId) {
     const taskData = this.taskFields;
-    if (taskData.status === COMPLETED) {
-      taskData.percentCompleted = 100;
-    }
     const cleanBody = this.cleanReqBody(taskData);
     if (taskData.status || taskData.percentCompleted) {
       try {
+        this.isLoading = true;
```

Comment:
Do we need this in try? Try should contain the minimal section of code that it can.